grc/gui/ActionHandler.py

```diff
@@ -200,133 +200,124 @@ def flow_graph_update(fg=flow_graph):
-                                if flow_graph.get_block(source_id) not in flow_graph.get_selected_blocks():
-                                    pads.append({'key': connection.get_sink().get_key(), 'coord': connection.get_source().get_coordinate(), 'block_id' : block.get_id(), 'direction': 'source'})
-
-                                if flow_graph.get_block(sink_id) not in flow_graph.get_selected_blocks():
-                                    pads.append({'key': connection.get_source().get_key(), 'coord': connection.get_sink().get_coordinate(), 'block_id' : block.get_id(), 'direction': 'sink'})
-
-
-                        # Copy the selected blocks and paste them into a new page
-                        #   then move the flowgraph to a reasonable position
-                        Actions.BLOCK_COPY()
-                        main.new_page()
-                        Actions.BLOCK_PASTE()
-                        coords = (x_min,y_min)
-                        flow_graph.move_selected(coords)
-
-
-                        # Set flow graph to heir block type
-                        top_block  = flow_graph.get_block("top_block")
-                        top_block.get_param('generate_options').set_value('hb')
-
-                        # this needs to be a unique name
-                        top_block.get_param('id').set_value('new_heir')
-
-                        # Remove the default samp_rate variable block that is created
-                        remove_me  = flow_graph.get_block("samp_rate")
-                        flow_graph.remove_element(remove_me)
-
-
-                        # Add the param blocks along the top of the window
-                        x_pos = 150
-                        for param in params:
-                            param_id = flow_graph.add_new_block('parameter',(x_pos,10))
-                            param_block = flow_graph.get_block(param_id)
-                            param_block.get_param('id').set_value(param)
-                            x_pos = x_pos + 100
-
-                        for pad in pads:
-                            # Add the pad sources and sinks within the new heir block
-                            if pad['direction'] == 'sink':
-
-                                # Add new PAD_SINK block to the canvas
-                                pad_id = flow_graph.add_new_block('pad_sink', pad['coord'])
-
-                                # setup the references to the sink and source
-                                pad_block = flow_graph.get_block(pad_id)
-                                pad_sink = pad_block.get_sinks()[0]
-
-                                source_block = flow_graph.get_block(pad['block_id'])
-                                source = source_block.get_source(pad['key'])
-
-                                # Ensure the port types match
-                                while pad_sink.get_type() != source.get_type():
```

Comment:
I removed these while loops and the type_controller_modify() calls. Not sure how correct this is.